cron-jobs/check_archlinux/parse_pkgbuilds.sh

```diff
@@ -10,21 +10,12 @@ variables=('pkgname' 'pkgbase' 'epoch' 'pkgver' 'pkgrel' 'makedepends' 'arch' ${
 readonly -a variables splitpkg_overrides
 
 backup_package_variables() {
-	for var in ${splitpkg_overrides[@]}; do
-		indirect="${var}_backup"
-		eval "${indirect}=(\${$var[@]})"
-	done
+	restore_package_variables=$(declare -p "${splitpkg_overrides[@]}" 2>/dev/null || true)
 }
 
 restore_package_variables() {
-	for var in ${splitpkg_overrides[@]}; do
-		indirect="${var}_backup"
-		if [ -n "${!indirect}" ]; then
-			eval "${var}=(\${$indirect[@]})"
-		else
-			unset ${var}
-		fi
-	done
+	unset "${splitpkg_overrides[@]}"
+	eval "$restore_package_variables"
```

Comment:
FWIW: https://lists.archlinux.org/pipermail/pacman-dev/2018-March/022406.html

TBH this whole script seems somewhat convoluted. If anything I think I'd rather port `check_packages.py` to use pyalpm instead of ctypes, and `makepkg --printsrcinfo` + [python-srcinfo](https://github.com/kyrias/python-srcinfo) rather than `parse_pkgbuilds.sh`

But then again, I don't think anyone still uses it...

Reply:
Agree about it being convoluted.

Isn't your solution going to have `declare -p` spit things on stderr?